Added flightComputer.cpp, README.md, SensorData.cpp, StateMachine.cpp and StaticVariables.cpp

EricC1231 opened the pull request:
-flightComputer, Sensordata, and StateMachine were put into classes 
-main template was implemented for FlightComputer, and StateMachine.

EricC1231 committed:
Added flightComputer.cpp, README.md, SensorData.cpp, StateMachine.cpp, and StaticVariables.cpp
-flightComputer, Sensordata, and StateMachine were put into classes
-main template was implemented for Flightcomputer, and StateMachine.

frroossst commented on Software/flight_computer_program/FlightComputer.cpp:
```diff
@@ -0,0 +1,47 @@
+class FlightComputer{
+    bool AbleToFly = true;
+
+    StateMachine flight_function = StateMachine();
```
Don't initialise class members like this, do it in a constructor

frroossst commented on Software/flight_computer_program/StateMachine.cpp:
```diff
@@ -0,0 +1,80 @@
+        sensor_data = data;
+    }
+
+    enum FlightStates{
```
keep typedefs separate from class definitions

EricC1231 replied:
I don't quite understand what you mean here?

frroossst replied:
FlightStates is a type definition correct? It defines type of enum, you usually put those in a separate header `.h` or `.hpp` file. So that they can be used by other classes

frroossst commented on Software/flight_computer_program/FlightComputer.cpp:
```diff
@@ -0,0 +1,47 @@
+
+    StateMachine flight_function = StateMachine();
+
+    SensorData sensor_data = SensorData();
```
same goes for this, don't initialise class member, set in a constructor

EricC1231 committed:
added header files for every class and redesigned the classes to acomidate that.

EricC1231 commented:
I added a second commit, hopefully it solves the issues.

bristinemuluh committed:
Update StateMachine.h

Moved the flight states enum to it's own header file

bristinemuluh committed:
Update StateMachine.h

bristinemuluh committed:
Create flight_states.h

bristinemuluh committed:
Update StateMachine.cpp

Added initialization of flight_state to the constructor

bristinemuluh committed:
Update FlightComputer.h

Added constructor, changed AbleToFly to snake case. Moving initialization of able_to_fly to the constructor

bristinemuluh committed:
Update FlightComputer.h

Removed void return type from constructor

bristinemuluh committed:
Update FlightComputer.cpp

Added definition of constructor

bristinemuluh committed:
Update FlightComputer.cpp

Added class member initializations in constructor

bristinemuluh reviewed (commented):
I added a few commits to your pull request to address the issues raised by Adhyan.
Can you put each individual class in it's own folder with the same name as the class name? Makes our project cleaner. You will need to update your local copy first to the the changes I made.

frroossst commented:
A folder per class? That seems excessive. There'll be at least a dozen classes (vast underestimation), so does that mean a dozen folders? Also can you imagine working in an IDE environment, with that many folders, with only one file in them???? 

Please don't focus on clean code, It is something touted by managers and non-programmers who have little to no experience writing code. 

Excessive abstraction is exponentially worse than no abstraction. 

[Clean code considered harmful](https://medium.com/@aronisfotis/why-clean-code-practices-can-be-harmful-a2ba141ca5df)

If not me then believe Ryan Dahl, the guy who made the nodeJS runtime: https://tinyclouds.org/rant
> The users - the one who just want to see a webpage - don't care. They don't care how we organize /usr, they don't care about zombie processes, they don't care about bash tab completion, they don't care if zlib is dynamically linked or statically linked to Node. There will come a point where the accumulated complexity of our existing systems is greater than the complexity of creating a new one. When that happens all of this shit will be trashed. 

> The only thing that matters in software is the experience of the user.

Please write shitty code that runs especially right now. You can always have a huge refactor further down the line.
